File: android/2048/src/main/java/com/dreampany/play2048/data/source/local/Pref.java
package com.dreampany.play2048.data.source.local;

import android.content.Context;

import com.dreampany.framework.data.source.pref.FramePref;

import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * Created by Hawladar Roman on 3/7/18.
 * Dreampany Ltd
 * dreampanymail@gmail.com
 */
@Singleton
public class Pref extends FramePref {
    
    @Inject
    Pref(Context context) {
        super(context);
    }
}

File: android/music/src/main/kotlin/com/dreampany/music/data/source/local/Pref.kt
package com.dreampany.music.data.source.local

import android.content.Context
import com.dreampany.frame.data.source.local.FramePref
import com.dreampany.music.misc.SortOrder
import javax.inject.Inject
import javax.inject.Singleton


/**
 * Created by Hawladar Roman on 6/25/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 */
@Singleton
class Pref @Inject constructor(context : Context) : FramePref(context) {

    val SONG_SORT_ORDER = "song_sort_order"

    fun getSongSortOrder(): String? {
        return getPublicString(SONG_SORT_ORDER, SortOrder.SongSortOrder.SONG_A_Z)
    }

}